Specs/DateTime/English/DateTimeModel.json

```diff
@@ -21171,6 +21171,118 @@
       }
     ]
   },
+  {
+    "Input": "I want to book a flight for the weekend of halloween",
+    "Context": {
+        "ReferenceDateTime": "2021-11-09T18:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "the weekend of halloween",
+        "Start": 28,
+        "End": 51,
+        "TypeName": "datetimeV2.daterange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "XXXX-10-31",
```

Comment:
Timex here is incorrect. It should mention it's a weekend ("WE"). And we should take care to use the right anchor date.
Also, what two resolutions are being returned here?
Shouldn't they be the weekend before the holiday and the weekend after the holiday?
Or is it meant to be only weekends that include the holiday?

Reply:
Using the parsing for holidays as a reference:
When the input is the word "halloween" for example, two resolutions are returned: one for the previous halloween and the next halloween. This is the case for when the holiday is ambigious: "halloween 2021" would only resolve to the halloween of that year. 

A similar approach is implemented for weekends. The two resolutions are the two weekends corresponding to the previous and next instances of that holiday. (halloween 2021 weekend and halloween 2022 weekend).

As for the daterange itself, the resolved daterange itself extends from the holiday to the nearest weekend.

This is the behavior we thought was most appropriate, but it can be changed if you do not think it is correct.

Reply:
This has been updated, we can resolve the conversation if the new format is correct.